File: CS_NET Web App/CS_NET Web App/Controllers/LoginController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using CS_NET_Web_App.Models;
using System.Configuration;

namespace CS_NET_Web_App.Controllers
{
  public class LoginController : Controller
  {
    // GET: Login
    public ActionResult Index()
    {
      return View(new LoginVM());
    }

    [HttpPost]
    public ActionResult Index(LoginVM pLogin)
    {
      var loginClient = new SBMemberSecuritySvc.MemberSecuritySoapClient("MemberSecuritySoap");
      SBMemberSecuritySvc.ValidatePasswordReturn validatePasswordReturn = loginClient.ValidatePassword(ConfigurationManager.AppSettings["WSToken"], pLogin.Username, -1, pLogin.Password);

      switch(validatePasswordReturn.ReturnCode)
      {
        case 0:
          var memberStatusClient = new SBMemberSvc.MemberSoapClient("MemberSoap");
          var memberStateReturn = memberStatusClient.FetchMemberState(ConfigurationManager.AppSettings["WSToken"], pLogin.Username, -1);

          if (memberStateReturn.ReturnCode == 0 && memberStateReturn.MemberIsSuspended != -1)
          {
            Session.Add("memberId", pLogin.Username);
            return RedirectToAction("Index", "Member");
          } else
          {
            ViewBag.Errormessage = "There was an error or the member account is suspended. Please contact your system administrator.";
          }
          break;
        case 1:
          ViewBag.ErrorMessage = "Undefined Error";
          break;
        case 2:
          ViewBag.ErrorMessage = "Required field missing";
          break;
        case 100:
          ViewBag.ErrorMessage = "Invalid security token";
          break;
        case 101:
          ViewBag.ErrorMessage = "Insufficient Permission";
          break;
        case 102:
          ViewBag.ErrorMessage = "Security Token Suspended";
          break;
        case 5000:
          ViewBag.ErrorMessage = "Member Not Found";
          break;
        case 15000:
          ViewBag.ErrorMessage = "Invalid Password";
          break;
        default:
          break;
      }
      return View(pLogin);
    }

    public ActionResult Logoff()
    {
      Session.Clear();
      return RedirectToAction("Index");
    }
  }
}

File: VB_NET Web App/VB_NET Web App/Controllers/LoginController.vb
Imports System.Web.Mvc
Imports System.Configuration

Namespace Controllers
    Public Class LoginController
        Inherits Controller

    ' GET: Login
    Function Index() As ActionResult
      Return View(New LoginVM)
    End Function

    ' POST: Login
    <HttpPost>
    Function Index(ByVal pLogin As LoginVM) As ActionResult
      Dim loginClient = New SBMemberSecuritySvc.MemberSecuritySoapClient("MemberSecuritySoap")
      Dim validatePasswordReturn As SBMemberSecuritySvc.ValidatePasswordReturn = loginClient.ValidatePassword(ConfigurationManager.AppSettings("WSToken"), pLogin.Username, -1, pLogin.Password)
      Select Case validatePasswordReturn.ReturnCode
        Case 0
          Dim memberStatusClient = New SBMemberSvc.MemberSoapClient("MemberSoap")
          Dim memberStateReturn As SBMemberSvc.FetchMemberStateReturn = memberStatusClient.FetchMemberState(ConfigurationManager.AppSettings("WSToken"), pLogin.Username, -1)
          If memberStateReturn.ReturnCode = 0 And Not memberStateReturn.MemberIsSuspended = True Then
            Session.Add("memberId", pLogin.Username)
            Return RedirectToAction("Index", "Member")
          Else
            ViewBag.ErrorMessage = "There was an error or the member account is suspended. Please contact your system administrator."
          End If
        Case 1
          ViewBag.ErrorMessage = "Undefined Error"
        Case 2
          ViewBag.ErrorMessage = "Required field missing"
        Case 100
          ViewBag.ErrorMessage = "Invalid security token"
        Case 101
          ViewBag.ErrorMessage = "Insufficient Permission"
        Case 102
          ViewBag.ErrorMessage = "Security Token Suspended"
        Case 5000
          ViewBag.ErrorMessage = "Member Not Found"
        Case 15000
          ViewBag.ErrorMessage = "Invalid Password"
        Case Else
          ViewBag.ErrorMessage = "Unknown Return Code"
      End Select
      Return View(pLogin)
    End Function

    Function Logoff() As ActionResult
      Session.Clear()
      Return RedirectToAction("Index")
    End Function
  End Class
End Namespace